src/rules/noAccessStateInSetstateRule.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * @license
+ * Copyright 2018 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Lint from "tslint";
+import { isCallExpression, isClassDeclaration, isPropertyAccessExpression } from "tsutils";
+import * as ts from "typescript";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "no-access-state-in-setstate",
+        description: "Reports usage of this.state within setState",
+        rationale: Lint.Utils.dedent
+            `Usage of this.state might result in errors when two state calls are \
+            called in batch and thus referencing old state and not the current state.`,
```

Comment:
It would be good to reference [these react docs](https://reactjs.org/docs/react-component.html#setstate) in this rationale. you can use markdown

Reply:
Added a link to the docs. Is the phrasing alright? (English being not my first language and all)